[2020-12-22 14:17:41,098] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:17:41,112] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:17:41,115] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:17:41,116] {taskinstance.py:867} INFO - Starting attempt 45 of 48
[2020-12-22 14:17:41,116] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:17:41,130] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:17:41,135] {standard_task_runner.py:53} INFO - Started process 60522 to run task
[2020-12-22 14:17:41,195] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:17:41,222] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,221] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:17:41,232] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,231] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:17:41,234] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,234] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:17:41,268] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,268] {batch.py:166} INFO - Batch successfully submitted with id = 41.
[2020-12-22 14:17:41,269] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,269] {batch.py:80} INFO - Getting batch 41 status...
[2020-12-22 14:17:41,279] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,278] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:17:41,281] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,281] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/41
[2020-12-22 14:17:41,292] {logging_mixin.py:112} INFO - [2020-12-22 14:17:41,291] {batch.py:90} INFO - Batch 41 has not finished yet (state is 'starting')
[2020-12-22 14:18:01,314] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,314] {batch.py:80} INFO - Getting batch 41 status...
[2020-12-22 14:18:01,321] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,321] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:18:01,322] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,321] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/41
[2020-12-22 14:18:01,331] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,331] {batch.py:299} INFO - --------------------------------------------------Full log for batch 41--------------------------------------------------
[2020-12-22 14:18:01,339] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,339] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:18:01,341] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,341] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/41/log?from=0&size=100
[2020-12-22 14:18:01,350] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,350] {batch.py:309} INFO - stdout: 
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,352] {batch.py:309} INFO - 2020-12-22 14:17:43,107 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,353] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,353] {batch.py:309} INFO -   File "/tmp/spark-5963fb92-6bc1-4792-9df8-a888dad91dc0/main.py", line 87, in <module>
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,353] {batch.py:309} INFO -     main()
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,353] {batch.py:309} INFO -   File "/tmp/spark-5963fb92-6bc1-4792-9df8-a888dad91dc0/main.py", line 59, in main
[2020-12-22 14:18:01,353] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,353] {batch.py:309} INFO -     for filename in listdir(ROOT_DIR)
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - NameError: name 'listdir' is not defined
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 2020-12-22 14:17:46,493 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 2020-12-22 14:17:46,493 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-5963fb92-6bc1-4792-9df8-a888dad91dc0
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 2020-12-22 14:17:46,499 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 2020-12-22 14:17:46,500 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:18:01,354] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 2020-12-22 14:17:46,500 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:18:01,355] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,354] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:18:01,355] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,355] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:18:01,355] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,355] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:18:01,355] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,355] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 41--------------------------------------------------
[2020-12-22 14:18:01,355] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,355] {batch.py:335} INFO - Closing batch with id = 41
[2020-12-22 14:18:01,363] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,363] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:18:01,364] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,364] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/41
[2020-12-22 14:18:01,372] {logging_mixin.py:112} INFO - [2020-12-22 14:18:01,372] {batch.py:340} INFO - Batch 41 has been closed
[2020-12-22 14:18:01,381] {taskinstance.py:1128} ERROR - Batch 41 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 41 failed with state 'dead'
[2020-12-22 14:18:01,383] {taskinstance.py:1151} INFO - Marking task as UP_FOR_RETRY
[2020-12-22 14:18:06,142] {logging_mixin.py:112} INFO - [2020-12-22 14:18:06,142] {local_task_job.py:103} INFO - Task exited with return code 1
